[2.0] fix CI and updated 2.0 whats new

nabobalis opened the pull request:
Fixes #4911

dstansby commented on docs/whatsnew/2.0.rst:
```diff
@@ -10,7 +10,29 @@ Overview
-
+On this page, you can read about some of the big changes in this release:
+
+* :ref:`whatsnew-2.0-python`
+* :ref:`whatsnew-2.0-search`
+* :ref:`whatsnew-2.0-aiaprep`
+* :ref:`whatsnew-2.0-pixelindex`
+* :ref:`whatsnew-2.0-standards`
+* :ref:`whatsnew-2.0-overview`
+* :ref:`whatsnew-2.0-coordframe`
+* :ref:`whatsnew-2.0-carrington`
+* :ref:`whatsnew-2.0-proxy`
+* :ref:`whatsnew-2.0-citation`
```
Why not just add a local table of contents?

nabobalis replied:
Good question.

dstansby replied:
Is the plan to leave it like this, or add a contents?

nabobalis replied:
Yes.

dstansby reviewed (approved):
Seems fine - is there a reason we haven't got rid of the devedeps build on 2.0 yet? (if it's just no-ones bothered to yet, I can put in a PR)

nabobalis commented:
> Seems fine - is there a reason we haven't got rid of the devedeps build on 2.0 yet? (if it's just no-ones bothered to yet, I can put in a PR)

Devdeps are still important to check. Its broken to a test returning the wrong type but I am unsure why its only for 2.0 and not 2.1

nabobalis commented:
Online fails are real due to my changes.

ayshih commented:
> Devdeps are still important to check. Its broken to a test returning the wrong type but I am unsure why its only for 2.0 and not 2.1

2.1 falls too (see daily cron)

pep8speaks commented:
Hello @nabobalis! Thanks for updating this PR. We checked the lines you've touched for [PEP 8](https://www.python.org/dev/peps/pep-0008) issues, and found:





























* In the file [`sunpy/net/hek2vso/hek2vso.py`](https://github.com/sunpy/sunpy/blob/e64036d769752c16c95700ff42ed49208c3fa074/sunpy/net/hek2vso/hek2vso.py):

> [Line 50:101](https://github.com/sunpy/sunpy/blob/e64036d769752c16c95700ff42ed49208c3fa074/sunpy/net/hek2vso/hek2vso.py#L50): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (320 > 100 characters)
> [Line 52:101](https://github.com/sunpy/sunpy/blob/e64036d769752c16c95700ff42ed49208c3fa074/sunpy/net/hek2vso/hek2vso.py#L52): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (454 > 100 characters)
> [Line 83:101](https://github.com/sunpy/sunpy/blob/e64036d769752c16c95700ff42ed49208c3fa074/sunpy/net/hek2vso/hek2vso.py#L83): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (134 > 100 characters)
> [Line 87:101](https://github.com/sunpy/sunpy/blob/e64036d769752c16c95700ff42ed49208c3fa074/sunpy/net/hek2vso/hek2vso.py#L87): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (316 > 100 characters)

* In the file [`sunpy/net/helio/parser.py`](https://github.com/sunpy/sunpy/blob/e64036d769752c16c95700ff42ed49208c3fa074/sunpy/net/helio/parser.py):

> [Line 91:101](https://github.com/sunpy/sunpy/blob/e64036d769752c16c95700ff42ed49208c3fa074/sunpy/net/helio/parser.py#L91): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (560 > 100 characters)

nabobalis commented on sunpy/net/dataretriever/sources/noaa.py:
```diff
@@ -31,13 +31,11 @@ class NOAAIndicesClient(GenericClient):
-         Start Time           End Time      Source  Instrument  Wavelength
-    ------------------- ------------------- ------ ------------ ----------
-    2016-01-01 00:00:00 2016-01-02 00:00:00   sdic noaa-indices        nan
+    0 Results from the VSOClient:
+    Start Time End Time Source Instrument Type
+    ---------- -------- ------ ---------- ----
     <BLANKLINE>
     <BLANKLINE>
```
This is odd.

nabobalis replied:
Fixed.

nabobalis commented on sunpy/net/dataretriever/sources/noaa.py:
```diff
@@ -72,7 +71,7 @@ def _can_handle_query(cls, *query):
         chklist = [x.__class__.__name__ in chkattr for x in query]
         for x in query:
-            if x.__class__.__name__ == 'Instrument' and x.value == 'noaa-indices':
+            if x.__class__.__name__ == 'Instrument' and x.value.lower() == 'noaa-indices':
```
WOOPS

nabobalis commented:
Bunch of things changed:

1. Added py39 wheels and test suite
2. Remembered we don't run the doctests properly in the online suite so I fixed that and any broken tests.
3. Found a bug in the NOAA-INDICES client.

dstansby commented on azure-pipelines.yml:
```diff
@@ -53,15 +53,14 @@ jobs:
-      - macos: py36
-        name: py36_test
-        posargs: -n=1
+      - macos: py38
+        name: py38_test
```
Why has this changed?

nabobalis replied:
Don't need it anymore.

nabobalis replied:
Oh you mean the Python version? We had too many 36 builds, I thought I would bump macos to a more modern version.

dstansby commented on azure-pipelines.yml:
```diff
@@ -53,15 +53,14 @@ jobs:
-      - linux: py38-online
-        name: py38_test_online
+      - linux: py39-online
+        name: py39_test_online
```
Adding python 3.9 support should get a changelog entry.

nabobalis replied:
Sure.

dstansby commented on setup.cfg:
```diff
@@ -100,6 +100,58 @@ remote_data_strict = True
+    ignore:direct construction of AsdfSchemaFile has been deprecated
+    ignore:direct construction of AsdfSchemaItem has been deprecated
+    ignore:direct construction of AsdfSchemaExampleItem has been deprecated
+    # See https://github.com/Cadair/parfive/issues/41
+    ignore:The loop argument is deprecated
+    # This is due to dependencies building with a numpy version different from
+    # the local installed numpy version, but should be fine
+    # See https://github.com/numpy/numpy/issues/15748#issuecomment-598584838
+    ignore:numpy.ufunc size changed:RuntimeWarning
+    ignore:numpy.ndarray size changed:RuntimeWarning
+    # See https://github.com/mvantellingen/python-zeep/issues/956
+    ignore:defusedxml.lxml is no longer supported:DeprecationWarning
+    # See https://github.com/aio-libs/aiohttp/issues/4842
+    ignore:"@coroutine" decorator is deprecated
+    # See https://github.com/spacetelescope/asdf/issues/789
+    ignore:direct construction of AsdfSchemaFile has been deprecated
+    ignore:direct construction of AsdfSchemaItem has been deprecated
+    ignore:direct construction of AsdfSchemaExampleItem has been deprecated
+    # See https://github.com/scikit-image/scikit-image/issues/4848
+    ignore:Converting `np.inexact` or `np.floating` to a dtype is deprecated
+    # See https://github.com/astropy/astropy/pull/10570
+    ignore:leap-second auto-update failed
+    # See https://github.com/astropy/extension-helpers/issues/23
+    ignore:Distutils was imported before Setuptools
+    # Figure tests use agg backend so show doesn't work but we use it in peek
+    ignore:Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.
+    # Ignore doctest issues: docs/guide/tour.rst
+    ignore:File may have been truncated*
+    # Some tests use data that have dates in the future and ERFA does not like.
+    ignore:ERFA function "d2dtf"*
+    ignore:ERFA function "dtf2d"*
+    # Currently helio makes unverified requests - this filter should be removed when
+    # https://github.com/sunpy/sunpy/issues/4401 is fixed
+    ignore:Unverified HTTPS request
+    # https://github.com/astropy/astropy/issues/11272
+    ignore:elementwise == comparison failed and returning scalar instead:FutureWarning:astropy.table
+    # This isn't really an issue it's an annoyance, it's triggered by printing
+    # an astropy table inside a test, which is generally only done when
+    # debugging.
+    ignore:the imp module is deprecated in favour of importlib:DeprecationWarning:ipykernel.iostream
```
Why add these lines? We're still ignoring warnings on the 2.0.x branch.

nabobalis replied:
It came with a backport and I cba to remove it.

nabobalis replied:
Removed.

dstansby reviewed (changes requested):
As well as my individual comments, the changelog entries from everything that has been backported also need to be added to this PR.

nabobalis commented:
> As well as my individual comments, the changelog entries from everything that has been backported also need to be added to this PR.

If a PR that was backported had a changelog, it should have come with the backport.

nabobalis commented on setup.cfg:
```diff
@@ -100,6 +100,58 @@ remote_data_strict = True
+    ignore:direct construction of AsdfSchemaFile has been deprecated
+    ignore:direct construction of AsdfSchemaItem has been deprecated
+    ignore:direct construction of AsdfSchemaExampleItem has been deprecated
+    # See https://github.com/Cadair/parfive/issues/41
+    ignore:The loop argument is deprecated
+    # This is due to dependencies building with a numpy version different from
+    # the local installed numpy version, but should be fine
+    # See https://github.com/numpy/numpy/issues/15748#issuecomment-598584838
+    ignore:numpy.ufunc size changed:RuntimeWarning
+    ignore:numpy.ndarray size changed:RuntimeWarning
+    # See https://github.com/mvantellingen/python-zeep/issues/956
+    ignore:defusedxml.lxml is no longer supported:DeprecationWarning
+    # See https://github.com/aio-libs/aiohttp/issues/4842
+    ignore:"@coroutine" decorator is deprecated
+    # See https://github.com/spacetelescope/asdf/issues/789
+    ignore:direct construction of AsdfSchemaFile has been deprecated
+    ignore:direct construction of AsdfSchemaItem has been deprecated
+    ignore:direct construction of AsdfSchemaExampleItem has been deprecated
+    # See https://github.com/scikit-image/scikit-image/issues/4848
+    ignore:Converting `np.inexact` or `np.floating` to a dtype is deprecated
+    # See https://github.com/astropy/astropy/pull/10570
+    ignore:leap-second auto-update failed
+    # See https://github.com/astropy/extension-helpers/issues/23
+    ignore:Distutils was imported before Setuptools
+    # Figure tests use agg backend so show doesn't work but we use it in peek
+    ignore:Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.
+    # Ignore doctest issues: docs/guide/tour.rst
+    ignore:File may have been truncated*
+    # Some tests use data that have dates in the future and ERFA does not like.
+    ignore:ERFA function "d2dtf"*
+    ignore:ERFA function "dtf2d"*
+    # Currently helio makes unverified requests - this filter should be removed when
+    # https://github.com/sunpy/sunpy/issues/4401 is fixed
+    ignore:Unverified HTTPS request
+    # https://github.com/astropy/astropy/issues/11272
+    ignore:elementwise == comparison failed and returning scalar instead:FutureWarning:astropy.table
+    # This isn't really an issue it's an annoyance, it's triggered by printing
+    # an astropy table inside a test, which is generally only done when
+    # debugging.
+    ignore:the imp module is deprecated in favour of importlib:DeprecationWarning:ipykernel.iostream
```
```suggestion
```

dstansby commented:
> > As well as my individual comments, the changelog entries from everything that has been backported also need to be added to this PR.
> 
> If a PR that was backported had a changelog, it should have come with the backport.

In that case could you add a list of all the PRs that have been backported to this PR description, just so we have a paper trail of which other PRs are being backported?

nabobalis commented:
> > > As well as my individual comments, the changelog entries from everything that has been backported also need to be added to this PR.
> > 
> > 
> > If a PR that was backported had a changelog, it should have come with the backport.
> 
> In that case could you add a list of all the PRs that have been backported to this PR description, just so we have a paper trail of which other PRs are being backported?

Its in the commit names?

dstansby commented:
Ah I didn't realise, 👍
